pkg/operator/controller/status/controller.go

```diff
@@ -270,21 +298,39 @@ func computeOperatorDegradedCondition(dns *operatorv1.DNS) configv1.ClusterOpera
 }
 
 // computeOperatorProgressingCondition computes the operator's current Progressing status state.
-func computeOperatorProgressingCondition(dnsAvailable bool, oldVersions, curVersions []configv1.OperandVersion,
-	operatorReleaseVersion, coreDNSImage, openshiftCLIImage, kubeRBACProxyImage string) configv1.ClusterOperatorStatusCondition {
-	// TODO: Update progressingCondition when an ingresscontroller
-	//       progressing condition is created. The Operator's condition
-	//       should be derived from the ingresscontroller's condition.
+func computeOperatorProgressingCondition(haveDNS bool, dns *operatorv1.DNS, oldVersions, curVersions []configv1.OperandVersion, operatorReleaseVersion, coreDNSImage, openshiftCLIImage, kubeRBACProxyImage string) configv1.ClusterOperatorStatusCondition {
 	progressingCondition := configv1.ClusterOperatorStatusCondition{
 		Type: configv1.OperatorProgressing,
 	}
+	status := configv1.ConditionUnknown
+	var messages, progressingReasons []string
 
-	progressing := false
-
-	var messages []string
-	if !dnsAvailable {
-		messages = append(messages, "DNS default unavailable")
-		progressing = true
+	if !haveDNS {
+		status = configv1.ConditionTrue
+		progressingReasons = append(progressingReasons, "DNSDoesNotExist")
+		messages = append(messages, `DNS "default" does not exist`)
+	} else {
+		foundProgressingCondition := false
+		for _, cond := range dns.Status.Conditions {
+			if cond.Type != operatorv1.OperatorStatusTypeProgressing {
+				continue
+			}
+			foundProgressingCondition = true
+			switch cond.Status {
+			case operatorv1.ConditionTrue:
+				status = configv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "DNSReportsProgressingIsTrue")
+				messages = append(messages, fmt.Sprintf("DNS %q reports Progressing=True: %q", dns.Name, cond.Message))
+			case operatorv1.ConditionUnknown:
+				progressingReasons = append(progressingReasons, "DNSReportsProgressingIsUnknown")
+				messages = append(messages, fmt.Sprintf("DNS %q reports Progressing=Unknown: %q", dns.Name, cond.Message))
+			}
+			break
+		}
+		if !foundProgressingCondition {
+			progressingReasons = append(progressingReasons, "DNSDoesNotReportsProgressingStatus")
```

Comment:
```suggestion
			progressingReasons = append(progressingReasons, "DNSDoesNotReportsProgressingStatus")
```

Is `reports` a typo?

Reply:
Fixed.  Thanks!